packages/wallet/src/components/indirect-funding/funding-step.tsx

```diff
@@ -63,80 +42,45 @@ export const fundingStepByState = (state: indirectFunding.IndirectFundingState):
   }
 };
 
-export enum Step {
-  WAIT_FOR_PRE_FUND_SETUP,
-  WAIT_FOR_DIRECT_FUNDING,
-  WAIT_FOR_POST_FUND_SETUP,
-  WAIT_FOR_LEDGER_UPDATE,
-  WAIT_FOR_CONSENSUS,
-}
-
-const message = (iconStep: Step, currentStep: Step) => {
-  switch (iconStep) {
-    case Step.WAIT_FOR_PRE_FUND_SETUP:
-      if (currentStep < iconStep) {
-        return 'Not ready for prefund setup';
-      } else if (currentStep === iconStep) {
-        return 'Waiting for opponent prefund setup';
-      } else {
-        return 'Received opponent prefund setup';
-      }
-    case Step.WAIT_FOR_DIRECT_FUNDING:
-      if (currentStep < iconStep) {
-        return 'Not ready for ledger direct funding';
-      } else if (currentStep === iconStep) {
-        return 'Waiting for ledger direct funding';
-      } else {
-        return 'Ledger channel directly funded';
-      }
-    case Step.WAIT_FOR_POST_FUND_SETUP:
-      if (currentStep < iconStep) {
-        return 'Not ready for postfund setup';
-      } else if (currentStep === iconStep) {
-        return 'Waiting for opponent postfund setup';
-      } else {
-        return 'Received opponent postfund setup';
-      }
-    case Step.WAIT_FOR_LEDGER_UPDATE:
-      if (currentStep < iconStep) {
-        return 'Not ready to allocate from ledger channel';
-      } else if (currentStep === iconStep) {
-        return 'Waiting for opponent ledger channel update';
-      } else {
-        return 'Received opponent ledger channel update';
-      }
-  }
-  return '';
-};
+const messagesForStepList: MessagesForStep[] = [
```

Comment:
Maybe a better option here is to have an object that has a key for every step in the `Step` enum on line 12. 

~You can get some type safety by defining a `MessagesForStep` interface that looks like `{ [step: Step]: MessageForStep }`~ Sadly, `Step` is not a valid type for an index signature.

If you really want type safety, you could have a `messagesForStepList` _function_, that has an exhaustive `switch` statement on the `Step` enum.

Reply:
I did try out the `messagesForStepList` function approach earlier and didn't think the additional complexity was worth the additional safety.